Backend/Repositories/DeckRepository.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Domain.Entities;
+using Infrastructure.Persistance.Relational;
+using Domain.DTOs;
+
+namespace Backend.Repositories
+{
+    //Recieves DTO looks for Entities
+    //Sends DTO's back
+    public class DeckRepository : IDeckRepository
+    {
+        private readonly RelationalContext _context;
+
+        public DeckRepository(RelationalContext context)
+        {
+            _context = context;
+        }
+
+        public DeckDTO AddDeck(DeckDTO deckDto)
+        {
+            var deck = new Deck
+            {
+                Name = deckDto.Name,
+                UserId = deckDto.UserId,
+                DeckCards = new List<DeckCard>()
+            };
+
+            foreach (var card in deckDto.Cards)
+            {
+                var deckCard = new DeckCard
+                {
+                    CardId = card.Id,
+                    Deck = deck
+                };
+                deck.DeckCards.Add(deckCard);
+            }
+
+            _context.Decks.Add(deck);
+            _context.SaveChanges();
+
+            return GetDeckById(deck.Id);
+        }
+
+        public void DeleteDeck(DeckDTO deckDto)
+        {
+            var deck = _context.Decks.Find(deckDto.Id);
+            if (deck != null)
+            {
+                var deckCards = _context.DeckCards.Where(dc => dc.DeckId == deck.Id);
+                foreach (var deckCard in deckCards)
+                {
+                    _context.DeckCards.Remove(deckCard);
+                }
+                _context.Decks.Remove(deck);
+                _context.SaveChanges();
+            }
```

Comment:
Metoden virker, men vi har ikke nogen god måde at se om den er blevet slettet. Hvis der kommer et ID ind som ikke stemmer overens med noget der er i databasen så sker der ikke noget. Det ville være rart at få et svar tilbage der indikerer om det gik godt

Reply:
Er det ikke det, som ligger i DeckControllerens MapDelete? der prøver den at hente fra db, og efterfølgende tjekker for om den lokale var er "null"

Reply:
I feel destroyed by facts and logic (that dont care about my feelings)